Build failed on macOS

I got these error when I run `make`:
```bash
/Users/thearas/repos/simple-sqlite/core/hash.c:101:12: error: implicit declaration of function 'sqliteStrNICmp' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    return sqliteStrNICmp((const char*)pKey1,(const char*)pKey2,n1);
           ^
/Users/thearas/repos/simple-sqlite/core/hash.c:101:12: note: did you mean 'sqliteStrDup'?
/Users/thearas/repos/simple-sqlite/core/util.h:5:7: note: 'sqliteStrDup' declared here
char *sqliteStrDup(const char *z);
      ^
/Users/thearas/repos/simple-sqlite/core/hash.c:118:12: error: implicitly declaring library function 'memcmp' with type 'int (const void *, const void *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
    return memcmp(pKey1,pKey2,n1);
           ^
/Users/thearas/repos/simple-sqlite/core/hash.c:118:12: note: include the header <string.h> or explicitly provide a declaration for 'memcmp'
/Users/thearas/repos/simple-sqlite/core/hash.c:173:14: error: implicitly declaring library function 'malloc' with type 'void *(unsigned long)' [-Werror,-Wimplicit-function-declaration]
    new_ht = malloc(new_size*sizeof(struct _ht));
             ^
/Users/thearas/repos/simple-sqlite/core/hash.c:173:14: note: include the header <stdlib.h> or explicitly provide a declaration for 'malloc'
/Users/thearas/repos/simple-sqlite/core/hash.c:174:5: error: implicitly declaring library function 'memset' with type 'void *(void *, int, unsigned long)' [-Werror,-Wimplicit-function-declaration]
    memset(new_ht, 0, new_size*sizeof(struct _ht));
    ^
/Users/thearas/repos/simple-sqlite/core/hash.c:174:5: note: include the header <string.h> or explicitly provide a declaration for 'memset'
/Users/thearas/repos/simple-sqlite/core/hash.c:325:9: error: implicitly declaring library function 'memcpy' with type 'void *(void *, const void *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
        memcpy((void*)new_elem->pKey, pKey, nKey);
        ^
/Users/thearas/repos/simple-sqlite/core/hash.c:325:9: note: include the header <string.h> or explicitly provide a declaration for 'memcpy'
2 warnings and 5 errors generated.
make[2]: *** [CMakeFiles/simple_sqlite.dir/core/hash.c.o] Error 1
make[1]: *** [CMakeFiles/simple_sqlite.dir/all] Error 2
make: *** [all] Error 2
```

### OS

```
Darwin ThearasdeMacBook-Pro-13.local 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:20 PDT 2021; root:xnu-7195.141.6~3/RELEASE_ARM64_T8101 arm64
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,11 @@ project(simple_sqlite C)
 
 set(CMAKE_C_STANDARD 11)
 
+# issue #1
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+        set(CMAKE_C_FLAGS "-Wno-implicit-function-declaration")
+endif()
+
 include_directories(core)
 
 add_executable(simple_sqlite

--- a/README.md
+++ b/README.md
@@ -15,7 +15,7 @@ Compile and running instructions for Unix
 4. Build with make
 <br/>`make`
 5. Run test 
-<br/>`./simple-sqlite`
+<br/>`./simple_sqlite`
 
 Use as your own Paging library
 ========

